### DISCUSSION TITLE
DayZ - Mods - Namalsk

### DISCUSSION THREAD

**Discussion by JacobOdell** [GameServerManagers/LinuxGSM#3980] [Q&A]
Hi guys, question for anyone that is familiar with DayZ modding. I am having some trouble with getting the Namalsk mod in particular to load. I have been able to successfully load another mod (GoreZ) but Namalsk is a slightly different beast as it is a different map/mission. 

## **The issue**
When I change "template" in the Dayz class within dayzserver/serverfiles/cfg/dayzserver.server.cfg:116 from "dayZOffline.chernarusplus" to "regular.namalsk", the server cannot boot. I see in the logs "No world with name namalsk" which you can see at dayzserver/log/console/dayzserver-console.log:1577

## **What I have done so far**
- Followed @PandaStudiosSWE's advice on #3899 and learned how to configure the startparameters and specify the mods folder.
- Successfully installed the GoreZ mod

- From their default configs, I have modified the following files:

```
dayzserver/lgsm/config-lgsm/dayzserver/dayzsever.cfg (added startparameters)
dayzserver/serverfiles/cfg/dayzserver.server.cfg (configured server to use regular.namalsk)
dayzserver/serverfiles/profiles (just created the dir, it was autopopulated with stuff)
dayzserver/serverfiles/keys (added keys from mods)
dayzserver/serverfiles/mods (create dir and put mods in there)
dayzserver/serverfiles/mpmissions (added regular.namalsk)
```

Here is my [dayzserver setup in full](https://drive.google.com/file/d/1Gjo6GJ6ARSQNIMW6qd6aX0jn8S2Xr7Pv/view?usp=sharing) so you can look for yourselves. Any advice is appreciated

Here is just the log
[dayzserver-console.log](https://github.com/GameServerManagers/LinuxGSM/files/9541701/dayzserver-console.log)

**Comment by MassCraxx** [comment]
Were you able to solve this issue? I am facing the same problem.

**Comment by MassCraxx** [comment]
As suggested by @JayC1337 [here](https://github.com/GameServerManagers/LinuxGSM/discussions/3899#discussioncomment-3817188) it is necessary to rename all files and folders of the mods to use only lowercase letters.

> Old DayZ modding problem which exists since the Arma 2 mod: You have to change every file and folder to lowercase in your mod directory. Easiest way: apt install rename. You can rename files inside a folder with rename 'y/A-Z/a-z/' *

**Comment by MassCraxx** [comment]
Unfortunately, even after renaming the files, my client still freezes on connecting to the server with the same mods selected that are used on the server. Connecting to other Experimental Servers running Namalsk works fine. 
I get some alarming server logs at the end of server startup:
```
19:22:18 [Script] scripts/3_Game/undergroundarealoader.c:83 | SpawnAllTriggerCarriers | [WARNING] :: [UndergroundAreaLoader GetData()] :: file not found in MISSION folder, your path is $mission:cfgundergroundtriggers.json Attempting DATA folder
19:22:18 [Script] scripts/3_Game/undergroundarealoader.c:83 | SpawnAllTriggerCarriers | [WARNING] :: [UndergroundAreaLoader GetData()] ::file not found in DATA folder, your path is DZ/worlds/namalsk/ce/cfgundergroundtriggers.json
19:22:18 String "STR_DN_UNKNOWN" not found (not in localization table)
19:22:18 String "STR_DN_UNKNOWN" not found (not in localization table)
19:22:18 String "STR_DN_UNKNOWN" not found (not in localization table)
19:22:18 Warning: No components in nst\ns_dayz\gear\lore\paper_files.p3d:geometry
19:22:18 [Script] scripts/5_Mission/mission\missionserver.c:89 | OnMissionStart | [WARNING] :: [EffectAreaLoader CreateZones] :: No contaminated area file found in MISSION folder, your path is $mission:cfgEffectArea.json Attempting DATA folder
19:22:18 [Script] scripts/5_Mission/mission\missionserver.c:89 | OnMissionStart | [WARNING] :: [EffectAreaLoader CreateZones] :: No contaminated area file found in DATA folder, your path is DZ/worlds/namalsk/ce/cfgEffectArea.json
19:22:18 Mission script has no main function, player connect will stay disabled!
```

Especially _Mission script has no main function_ tells me there is still something very wrong. The server log is attached. Any ideas?

[DayZServer_2022-10-08_19-22-08.RPT.log](https://github.com/GameServerManagers/LinuxGSM/files/9740169/DayZServer_2022-10-08_19-22-08.RPT.log)

**Comment by JayC1337** [comment]
Can you post your dayzserver.cfg in ~/lgsm/config-lgsm/dayzserver where you add the mods? I had to write it like this because the ; was interpreted wrong: mods="2405175615';'1559212036';'1582756848';'2865478912"

**Comment by MassCraxx** [comment]
I was using:
```
startparameters="-ip=${ip} -port=${port} -config=${servercfgfullpath} -profiles=profiles '-mod=mods/@NamalskSurvival;mods/@NamalskIsland;' -limitFPS=60 -dologs -adminlog -freezeCheck"
```

now I tried your syntax:
```
mods="mods/@NamalskSurvival';'mods/@NamalskIsland"
startparameters="-ip=${ip} -port=${port} -config=${servercfgfullpath} -profiles=profiles -mod=${mods} -limitFPS=60 -dologs -adminlog -freezeCheck"
```

I also tried rename the mod folders to their workshop id and moving them to /serverfiles,  and used `mods="2289461232';'2289456201"` but all produces the same result.

**Comment by JayC1337** [comment]
Going to test it on my server later to see if I can reproduce it. Just to be sure: Did you edit your ~/Serverfiles/cfg/dayzserver.server.cfg at the bottom to call the Namalsk mission file?

**Reply by JayC1337** [reply]
OK, installed the server and its working. Did not test via client as i'm still in bed LUL.
Here is what i did:
1. Created a new user
2. Installed via ./dayzserver install
3. Downloaded Namalsk Island via steamcmd: workshop_download_item 221100 2289456201
4. Downloaded Namalsk Survival via steamcmd: workshop_download_item 221100 2289461232
5. Moved Namalsk Island to serverfiles: mv /home/namalsk/.local/share/Steam/steamapps/workshop/content/221100/2289456201 serverfiles/
6. Moved Namalsk Survival to serverfiles: mv /home/namalsk/.local/share/Steam/steamapps/workshop/content/221100/2289461232 serverfiles/
7. Renamed containing files and folders in both mod folders to lower case: rename 'y/A-Z/a-z/' *
8. moved keys from Namlask files to serverfiles/keys directory
9. moved mpmission directory from 2289461232/extras/regular to serverfiles/mpmissions
10. edited serverfiles/cfg/dayzserver.server.cfg at the bottom to load correct namalsk mission file:
```
// Mission to load on server startup. <MissionName>.<TerrainName>
class Missions
{
    class DayZ
    {
        template="regular.namalsk";
    };
};
```
11. Edited /home/namalsk/lgsm/config-lgsm/dayzserver/dayzserver.cfg to following content:
```
##################################
####### Instance Settings ########
##################################
# PLACE INSTANCE SETTINGS HERE
## These settings will apply to a specific instance.
mods="2289456201';'2289461232"
```
Let me know if this helps.

**Comment by MassCraxx** [comment]
Thank you so much for your detailed description and your patience. I simply did not know I have to move the mission files to mpmission folder, I assumed it would read it from the mod folder directly.

Moving namalsk to the mpmission folder solved all remaining problems, all seems to be working prefectly now.

**Reply by JayC1337** [reply]
Great! No problem, hosting DayZ can be painful if you are new to it. If you have any questions feel free to ask.

**Comment by abzoabzoabzo** [comment]
Hello bro, i had the same problem and fixed it
You edit your file ''start''
make sure you have a -   before mod... like this 
the line start must be this 
start "DayZ Server" /min "DayZServer_x64.exe" -config=%serverConfig% -port=%serverPort% "-profiles=config" "-mod=@Namalsk Island;@Namalsk Survival" -cpuCount=%serverCPU% -dologs -adminlog -netlog -freezecheck



TOTAL is
@echo off
:start
::Server name (This is just for the bat file)
set serverName=SSG Namalsk Test Server
::Server files location
set serverLocation="G:\SteamLibrary\steamapps\common\DayZServer"
::Server Port
set serverPort=2302
::Server config
set serverConfig=serverDZ.cfg
::Logical CPU cores to use (Equal or less than available)
set serverCPU=2
::Sets title for terminal (DONT edit)
title %serverName% batch
::DayZServer location (DONT edit)
cd "%serverLocation%"
echo (%time%) %serverName% started.
::Launch parameters (edit end: -config=|-port=|-profiles=|-doLogs|-adminLog|-netLog|-freezeCheck|-filePatching|-BEpath=|-cpuCount=)
start "DayZ Server" /min "DayZServer_x64.exe" -config=%serverConfig% -port=%serverPort% "-profiles=config" "-mod=@Namalsk Island;@Namalsk Survival" -cpuCount=%serverCPU% -dologs -adminlog -netlog -freezecheck
::Time in seconds before kill server process (14400 = 4 hours)
timeout 14390
taskkill /im DayZServer_x64.exe /F
::Time in seconds to wait before..
timeout 10
::Go back to the top and repeat the whole cycle again
goto start

(you change location folder and server name but keep the rest as it is for the mods first to see, and then edit for more mod, gl)

**Reply by airtonix** [reply]
This is Linux Game Server, you're running windows. you're on the wrong discussion thread.


also to avoid confusion when illustrating code, wrap all code in three backticks: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code
